UI/obs-app.cpp

```diff
@@ -1098,10 +1098,75 @@ void OBSApp::ParseExtraThemeData(const char *path)
 	setPalette(pal);
 }
 
-bool OBSApp::SetTheme(std::string name, std::string path)
+bool OBSApp::ParseThemeMeta(const char *path)
 {
-	theme = name;
+	BPtr<char> data = os_quick_read_utf8_file(path);
+	CFParser cfp;
+	int ret;
+
+	cf_parser_parse(cfp, data, path);
+
+	themeMeta = nullptr;
```

Comment:
Not really a fan of this side effect. Would be cleaner to return the OBSThemeMeta object instead of a bool, and assign externally, i.e.: `themeMeta = ParseThemeMeta(path);`

Reply:
Done.